File: src/main/java/ast/ASTLambdaExpr.java
package ast;

public class ASTLambdaExpr extends ASTExpr {
    ASTFuncType type;
    ASTNodeArray<ASTNode> body;

    public ASTLambdaExpr(ASTFileLocation loc, ASTFuncType type, ASTNodeArray<ASTNode> body){
        super(loc);
        this.type = type;
        this.body = body;
    }

    public void print(int i){
        printIndent(i);
        System.out.println("ASTLambdaExpr: (type, body)");
        type.print(i+1);
        body.print(i+1);
    }

}

File: src/main/kotlin/ast/ASTLambdaExpr.kt
package ast

import compiler.genUniqueId

class ASTLambdaExpr(loc: ASTFileLocation, val type: ASTFuncType, val body: ASTNodeArray<ASTNode>) : ASTExpr(loc) {

    var enclosing_func: ASTNodeArray<ASTNode>? = null
    val id_num = genUniqueId()

    override fun print(i: Int) {
        printIndent(i)
        println("ASTLambdaExpr: (type, body)")
        type.print(i + 1)
        body.print(i + 1)
    }

}